lib/vtls/wolfssl.c

```diff
@@ -1008,6 +1008,45 @@ wolfssl_connect_step1(struct Curl_cfilter *cf, struct Curl_easy *data)
   return CURLE_OK;
 }
 
+static size_t wolfssl_version(char *buf, size_t size)
+{
+#if LIBWOLFSSL_VERSION_HEX >= 0x03006000
+  return msnprintf(buf, size, "wolfSSL/%s", wolfSSL_lib_version());
+#elif defined(WOLFSSL_VERSION)
+  return msnprintf(buf, size, "wolfSSL/%s", WOLFSSL_VERSION);
+#else
+  #error "WOLFSSL_VERSION must be defined"
+#endif
+}
+
+
+static char *wolfssl_strerror(unsigned long error, char *buf, size_t size)
+{
+  size_t len;
+  DEBUGASSERT(size);
+  *buf = '\0';
+
+  len = wolfssl_version(buf, size);
+  DEBUGASSERT(len < (size - 2));
+  if(len < (size - 2)) {
+    buf += len;
+    size -= (len + 2);
+    *buf++ = ':';
+    *buf++ = ' ';
+    *buf = '\0';
+  }
```

Comment:
Adding the library name plus version number in the error string seems... unorthodox. I don't think we should do it here as we don't do it for other backends.

Reply:
> Adding the library name plus version number in the error string seems... unorthodox. I don't think we should do it here as we don't do it for other backends.

This is done in [openssl](https://github.com/curl/curl/blob/master/lib/vtls/openssl.c#L928) backend so I wanted to remain consistent.

Reply:
actually I think openssl implementation is buggy as `buf` is incremented after adding a prefix and that's what we return so the version is never actually printed.